usecases/scenarios/scenario_publisher.go

```diff
@@ -99,6 +100,20 @@ func (publisher ScenarioPublisher) PublishOrUnpublishIteration(
 	return scenarioPublications, nil
 }
 
+func (publisher ScenarioPublisher) shutDownTestRunIfNeeded(ctx context.Context, tx repositories.Transaction, liveVersionId string) error {
+	testrun, errTestrun := publisher.ScenarioTestRunRepository.GetTestRunByLiveVersionID(ctx, tx, liveVersionId)
+	if errTestrun != nil {
+		return errTestrun
+	}
+	if testrun != nil && testrun.Status == models.Up {
```

Comment:
also include testrun.Status == pending I think (it probably won't happen really, just for completeness)